pkg/controller/garbagecollector/metrics/metrics.go

```diff
@@ -33,13 +34,84 @@ var (
+		Subsystem:      GarbageCollectorControllerSubsystem,
+		Name:           "attempt_to_delete_queue_retry_since_seconds",
+		Help:           "How long in seconds an item has been retrying in attempt to delete workqueue.",
+		Buckets:        metrics.ExponentialBuckets(0.001, 10, 10),
+		StabilityLevel: metrics.ALPHA,
+	}
+	orphanQueueRetrySinceSecondsOpts = &metrics.HistogramOpts{
+		Subsystem:      GarbageCollectorControllerSubsystem,
+		Name:           "attempt_to_orphan_queue_retry_since_seconds",
+		Help:           "How long in seconds an item has been retrying in attempt to orphan workqueue.",
+		Buckets:        metrics.ExponentialBuckets(0.001, 10, 10),
+		StabilityLevel: metrics.ALPHA,
+	}
+
+	deleteQueueRetrySinceSecondsDesc = metrics.NewDesc(metrics.BuildFQName(deleteQueueRetrySinceSecondsOpts.Namespace, deleteQueueRetrySinceSecondsOpts.Subsystem, deleteQueueRetrySinceSecondsOpts.Name),
+		deleteQueueRetrySinceSecondsOpts.Help,
+		nil,
+		deleteQueueRetrySinceSecondsOpts.ConstLabels,
+		deleteQueueRetrySinceSecondsOpts.StabilityLevel,
+		deleteQueueRetrySinceSecondsOpts.DeprecatedVersion)
+	orphanQueueRetrySinceSecondsDesc = metrics.NewDesc(metrics.BuildFQName(orphanQueueRetrySinceSecondsOpts.Namespace, orphanQueueRetrySinceSecondsOpts.Subsystem, orphanQueueRetrySinceSecondsOpts.Name),
+		orphanQueueRetrySinceSecondsOpts.Help,
+		nil,
+		orphanQueueRetrySinceSecondsOpts.ConstLabels,
+		orphanQueueRetrySinceSecondsOpts.StabilityLevel,
+		orphanQueueRetrySinceSecondsOpts.DeprecatedVersion)
 )
 
 var registerMetrics sync.Once
 
 // Register registers GarbageCollectorController metrics.
-func Register() {
+func Register(attemptToDeleteMetrics QueueMetrics, attemptToOrphanMetrics QueueMetrics) {
 	registerMetrics.Do(func() {
 		legacyregistry.MustRegister(GarbageCollectorResourcesSyncError)
+		legacyregistry.CustomMustRegister(newGCMetricsCollector(attemptToDeleteMetrics, attemptToOrphanMetrics))
 	})
 }
+
+func newGCMetricsCollector(attemptToDeleteMetrics, attemptToOrphanMetrics QueueMetrics) metrics.StableCollector {
+	return &gcMetricsCollector{
+		attemptToDeleteMetrics: attemptToDeleteMetrics,
+		attemptToOrphanMetrics: attemptToOrphanMetrics,
+	}
+}
+
+type gcMetricsCollector struct {
+	metrics.BaseStableCollector
+	attemptToDeleteMetrics QueueMetrics
+	attemptToOrphanMetrics QueueMetrics
+}
+
+func (g *gcMetricsCollector) DescribeWithStability(ch chan<- *metrics.Desc) {
+	ch <- deleteQueueRetrySinceSecondsDesc
+	ch <- orphanQueueRetrySinceSecondsDesc
+}
+func (g *gcMetricsCollector) CollectWithStability(ch chan<- metrics.Metric) {
+	g.collect(ch, deleteQueueRetrySinceSecondsDesc, deleteQueueRetrySinceSecondsOpts, g.attemptToDeleteMetrics)
+	g.collect(ch, orphanQueueRetrySinceSecondsDesc, orphanQueueRetrySinceSecondsOpts, g.attemptToOrphanMetrics)
+}
+
+func (g *gcMetricsCollector) collect(ch chan<- metrics.Metric, desc *metrics.Desc, histogramOpts *metrics.HistogramOpts, qMetrics QueueMetrics) {
```

Comment:
had to change collection from using prometheus metric to the following construct, due to  

```
ERROR: Some files are importing packages under github.com/prometheus/* but are not allow-listed to do so.
See: https://github.com/kubernetes/kubernetes/issues/89267
```

But it is a bit cumbersome, since there has not been any use case for collecting histograms yet. I wonder if it would make sense to make changes to component-base/metrics to support this scenario.